agent/config/file_watcher.go

```diff
@@ -81,12 +83,25 @@ func (w *FileWatcher) Stop() error {
 	w.stopOnce.Do(func() {
 		w.cancel()
 		<-w.done
-		close(w.EventsCh)
 		err = w.watcher.Close()
 	})
 	return err
 }
 
+// Add a file to the file watcher
+// Add will lock the file watcher during the add
+func (w *FileWatcher) Add(filename string) error {
```

Comment:
I thought in the other PR we decided to not need add/remove here, in favor of passing them all in at construction time?

Reply:
I needed to revise that decision because I did not consider the case where a user change the path to the cert files in the config. In that case we need to remove the old cert files from the watcher and add the new ones.